src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const modifiedStateCopy = [];
+  const stateCopy = { ...state };
+
+  actions.forEach(action => {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach(property => delete stateCopy[property]);
+        break;
+
+      case 'clear':
+        for (const property in stateCopy) {
```

Comment:
I'd rather call it `key`, property rather refer to key-value pair, it's something more abstract